src/gt4py/next/embedded/common.py

```diff
@@ -146,3 +146,31 @@ def _find_index_of_dim(
         if dim == d:
             return i
     return None
+
+
+def canonicalize_any_index_sequence(
+    index: common.AnyIndexSpec,
+) -> common.AnyIndexSpec:
+    new_index: common.AnyIndexSpec = (index,) if isinstance(index, slice) else index
+    if isinstance(index, slice):
+        new_index = _named_slice_to_named_range(index)
+    if isinstance(index, tuple) and all(isinstance(i, slice) for i in index):
+        new_index = tuple([_named_slice_to_named_range(i) for i in index])  # type: ignore[arg-type, assignment] # all i's are slices as per if statement
+    return new_index
+
+
+def _named_slice_to_named_range(idx: common.NamedSlice) -> common.NamedRange | common.NamedSlice:
```

Comment:
```suggestion
def _named_slice_to_named_range(idx: common.NamedSlice) -> common.NamedRange:
```

Reply:
Technically you could just be returning idx without any changes, so you do need `common.NamedSlice` in the return type